src/librbd/Operations.cc

```diff
@@ -779,7 +779,9 @@ int Operations<I>::snap_rollback(const cls::rbd::SnapshotNamespace& snap_namespa
   }
 
   r = prepare_image_update();
-  if (r < 0) {
+  if (r == -EAGAIN) {
```

Comment:
@jimmyway The 99.9999% use-case is that you received ```-EAGAIN``` for a real reason (the image is actively in-use) and a return code of ```-EROFS``` is appropriate. It seems you are trying to optimize for an edge case to the detriment of the common case.